zaimramlan/clean-swift-template
# Swiftlint Compliance and Improvements

# What's inside this PR?

- All
  - Added `typealias Models` to shorten call to each screen's models
- Business/Presentation/Display Protocols
  - Leave it to call the screen's model in full, to not unnecessary call a `typealias` from within Interactor, Presenter, ViewController respectively
- Models
  - Modified `AnalyticsEvents` to `typealias` to hint to refer to an centralised analytics constants enum
  - Declared `typealias Error = Error<ErrorType>` to simplify referencing to errors
  - Changed the comment header from `View Models` to `Types` to make more sense
  - Change `apiError` to `networkError` case to broaden the error scope
- Interactor
  - Simplified `fetchFromRemoteDataStore`, `trackAnalytics` and `perform___VARIABLE_sceneName___` use cases to be able to call shared workers directly, instead of having to wrap it through the screen's worker
  - Changed referencing screen worker as `lazy` as there's no specific reason to make it optional
- Router
  - Declared view controller as `weak` to prevent memory leaks
- Unit Tests
  - Updated both XCTest and Quick-Nimble templates to reflect the above changes

# How to manually test?

1. Install template using `$ make install_templates` command
1. On any Xcode project (that has the Quick & Nimble library installed)
    1. Create a new set of screen using the 'scene' template
    1. Create a new set of unit tests using both the XCTest & Quick-Nimble templates
1. Run unit test (CMD+U)
1. There should be no errors and all tests should pass
